Sources/SKCore/Configuration.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import Utility
+import PackageModel
+
+public protocol Configuration {
+  var build: SKCore.BuildConfiguration { get }
+}
+
+public protocol BuildConfiguration {
```

Comment:
What's the reasoning for using a protocol instead of just using a struct?

Reply:
The initial idea was to have generalized Configuration<BuildConfiguration> and use it here and in tests with specific types, but that's too much hassle after all.